site/app/libraries/database/DatabaseQueries.php

```diff
@@ -3178,6 +3178,25 @@ public function clearPeerGradingAssignments($gradeable_id) {
     public function insertPeerGradingAssignment($grader, $student, $gradeable_id) {
         $this->course_db->query("INSERT INTO peer_assign(grader_id, user_id, g_id) VALUES (?,?,?)", [$grader, $student, $gradeable_id]);
     }
+    
+    /**
+     * Adds an assignment for someone to grade another person for peer grading
+     *
+     * @param string $student
+     * @param string $grader
+     * @param string $gradeable_id
+     * @param string $feedback_full
+     * @param string $feedback_id
+     */
+    public function insertPeerGradingFeedback($grader, $student, $gradeable_id, $feedback_full, $feedback_id) {
+        $this->course_db->query("SELECT feedback_full FROM peer_feedback WHERE grader_id = ? AND user_id = ? AND g_id = ?", [$grader, $student, $gradeable_id]);
+        if (count($this->course_db->rows()) > 0) {
+            $this->course_db->query("UPDATE peer_feedback SET feedback_full = ?, feedback_id = ? WHERE grader_id = ? AND user_id = ? AND g_id = ?", [$feedback_full, $feedback_id, $grader, $student, $gradeable_id]);
+        }
+        else {
+            $this->course_db->query("INSERT INTO peer_feedback(grader_id, user_id, g_id, feedback_full, feedback_id) VALUES (?,?,?,?,?)", [$grader, $student, $gradeable_id, $feedback_full, $feedback_id]);
+        }
```

Comment:
I think you can simplify this logic by adding an `ON CONFLICT` clause to your `sql` command. That way, you don't need to run the `SELECT` first, just do the insert. Use the `ON CONFLICT` to do an update.